test/e2e/e2e-test/src/test/java/org/apache/skywalking/e2e/browser/BrowserE2E.java

```diff
@@ -0,0 +1,353 @@
+import org.apache.skywalking.e2e.service.instance.Instance;
+import org.apache.skywalking.e2e.service.instance.Instances;
+import org.apache.skywalking.e2e.service.instance.InstancesMatcher;
+import org.apache.skywalking.e2e.service.instance.InstancesQuery;
+import org.junit.jupiter.api.BeforeAll;
+import org.testcontainers.containers.ContainerState;
+import org.testcontainers.containers.DockerComposeContainer;
+
+import static java.util.Objects.nonNull;
+import static org.apache.skywalking.e2e.metrics.BrowserMetricsQuery.ALL_BROWSER_METRICS;
+import static org.apache.skywalking.e2e.metrics.BrowserMetricsQuery.ALL_BROWSER_PAGE_METRICS;
+import static org.apache.skywalking.e2e.metrics.BrowserMetricsQuery.ALL_BROWSER_PAGE_MULTIPLE_LINEAR_METRICS;
+import static org.apache.skywalking.e2e.metrics.BrowserMetricsQuery.ALL_BROWSER_SINGLE_VERSION_METRICS;
+import static org.apache.skywalking.e2e.metrics.MetricsMatcher.verifyMetrics;
+import static org.apache.skywalking.e2e.metrics.MetricsMatcher.verifyPercentileMetrics;
+import static org.apache.skywalking.e2e.utils.Times.now;
+import static org.apache.skywalking.e2e.utils.Yamls.load;
+
+@Slf4j
+@SkyWalkingE2E
+public class BrowserE2E extends SkyWalkingTestAdapter {
+
+    private static final int MAX_INBOUND_MESSAGE_SIZE = 1024 * 1024 * 50;
+
+    private static final String BROWSER_NAME = "e2e";
+
+    private static final String BROWSER_SINGLE_VERSION_NAME = "v1.0.0";
+
+    @SuppressWarnings("unused")
+    @DockerCompose({
+        "docker/browser/docker-compose.${SW_STORAGE}.yml",
+    })
+    protected DockerComposeContainer<?> compose;
+
+    @SuppressWarnings("unused")
+    @ContainerHostAndPort(name = "ui", port = 8080)
+    private HostAndPort swWebappHostPort;
+
+    @SuppressWarnings("unused")
+    @ContainerHostAndPort(name = "oap", port = 11800)
+    private HostAndPort oapHostPort;
+
+    @SuppressWarnings("unused")
+    @DockerContainer("oap")
+    private ContainerState oapContainer;
+
+    private RegisterGrpc.RegisterBlockingStub registerStub;
+
+    private BrowserPerfServiceGrpc.BrowserPerfServiceStub browserPerfServiceStub;
+
+    private volatile int browserId;
+
+    private volatile int browserSingleVersionId;
+
+    @BeforeAll
+    public void setUp() {
+        queryClient(swWebappHostPort);
+
+        final ManagedChannel channel =
+            NettyChannelBuilder.forAddress(oapHostPort.host(), oapHostPort.port())
+                               .nameResolverFactory(new DnsNameResolverProvider())
+                               .maxInboundMessageSize(MAX_INBOUND_MESSAGE_SIZE)
+                               .usePlaintext()
+                               .build();
+
+        registerStub = RegisterGrpc.newBlockingStub(channel);
+        browserPerfServiceStub = BrowserPerfServiceGrpc.newStub(channel);
+
+        register();
+        generateTraffic();
+    }
+
+    @RetryableTest
+    void verify() throws Exception {
```

Comment:
rename this, because the test framework will automatically convert the method name `camelCaseMethodName` to `verify camel case method name`